src/main/java/org/jenkinsci/plugins/ghprb/Ghprb.java

```diff
@@ -342,16 +333,24 @@ public static GhprbCause getCause(Run<?, ?> build) {
     }
     
 
-    public static GhprbTrigger extractTrigger(AbstractBuild<?, ?> build) {
-        return extractTrigger(build.getProject());
+    public static GhprbTrigger extractTrigger(Run<?, ?> build) {
+        return extractTrigger(build.getParent());
     }
 
-    public static GhprbTrigger extractTrigger(AbstractProject<?, ?> p) {
-        GhprbTrigger trigger = p.getTrigger(GhprbTrigger.class);
-        if (trigger == null || (!(trigger instanceof GhprbTrigger))) {
-            return null;
+    public static GhprbTrigger extractTrigger(Job<?, ?> p) {
+
+        ParameterizedJobMixIn.ParameterizedJob pJob = (ParameterizedJobMixIn.ParameterizedJob) p;
+        GhprbTrigger ghprbTrigger = null;
+        if (p instanceof ParameterizedJobMixIn.ParameterizedJob) {
```

Comment:
Isn't this guaranteed to be true based on line 342? I feel like this method might be clearer if it started off with the type check instead. e.g:

```java
public static GhprbTrigger extractTrigger(Job<?, ?> p) {
    if (!(p instanceof ParameterizedJobMixIn.ParameterizedJob)) {
        return null;
    }
    ParameterizedJobMixIn.ParemeterizedJob pjob = (ParameterizedJobMixIn.ParameterizedJob) p;
    ...
}